File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) == 0:
            return True
        
        m_s = {}
        m_t = {}
        for i in range(len(s)):
            if s[i] in m_s:
                if m_s[s[i]] != t[i]:
                    return False
            else:
                m_s[s[i]] = t[i]
            if t[i] in m_t:
                if m_t[t[i]] != s[i]:
                    return False
            else:
                m_t[t[i]] = s[i]    

        return True  
                
        

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int n = s.length();
        if (n == 1) {
            return true;
        }
        
        int[] mapS = new int[200];
        int[] mapT = new int[200];

        for(int i = 0; i < n; i++) {
            if(mapS[s.charAt(i)] != mapT[t.charAt(i)]) {
                return false;
            }
            mapS[s.charAt(i)] = i+1;
            mapT[t.charAt(i)] = i+1;
        }
        
        return true;
    }
}